uniffi_bindgen/src/library_mode.rs

```diff
@@ -42,24 +41,25 @@ pub fn generate_bindings<T: BindingGenerator + ?Sized>(
     out_dir: &Utf8Path,
     try_format_code: bool,
 ) -> Result<Vec<Component<T::Config>>> {
-    let cargo_metadata = MetadataCommand::new()
-        .exec()
-        .context("error running cargo metadata")?;
+    // TODO: we probably need this passed in somehow.
+    // Or can we also read it from `config_file_override` - what are the semantics of that?
+    #[cfg(feature = "cargo_metadata")]
+    let path_map = {
+        let cargo_metadata = cargo_metadata::MetadataCommand::new()
+            .exec()
+            .context("error running cargo metadata")?;
+        CratePathMap::from(cargo_metadata)
+    };
+    #[cfg(not(feature = "cargo_metadata"))]
+    let path_map = Default::default();
```

Comment:
How about extracting this to a `CratePathMap` method and defining that differently based on the feature flag?  It's hard to follow code where individual statements are behind feature flags.